### PR TITLE
EE-659: All the things added for the submission service

### PR DESCRIPTION
**Pull request by vinnicombe**

### CODE REVIEW

File: app/models/ETag.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json.{Format, Json}
+
+case class ETag(etag: String)
```
Comment (Pascoe121) [inline app/models/ETag.scala]: is an etag not an int?

File: app/services/TaiService.scala [hunk 2]
```diff
@@ -47,4 +50,24 @@ class TaiService @Inject()(taiConnector: TaiConnector) {
...
+  def updatePsubAmount(nino: String, year: Int, grossAmount: Int)
+               (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+
+    citizenDetailsConnector.getEtag(nino).flatMap {
+      response =>
+        response.status match {
+          case 200 =>
+            Json.parse(response.body).validate[ETag] match {
+              case JsSuccess(body, _) =>
+                taiConnector.updateProfessionalSubscriptionAmount(nino, year, body.etag.toInt, grossAmount)
```
Comment (Pascoe121) [inline app/services/TaiService.scala]: wouldn't need to int if we parse it out as an int

File: test/services/SubmissionServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  private val mockTaiService = mock[TaiService]
+  private val mockTaiConnector = mock[TaiConnector]
+  private val submissionService = new SubmissionService(mockTaiService, mockTaiConnector)
+
+  private val currentTaxYear = Seq(CurrentYear)
+  private val taxYearsWithCurrentYear = Seq(CurrentYear, CurrentYearMinus1)
+  private val taxYearsWithoutCurrentYear = Seq(CurrentYearMinus1, CurrentYearMinus2)
+  private val claimAmount = 100
+
+//  override def beforeEach() = reset(mockTaiConnector)
```
Comment (Pascoe121) [inline test/services/SubmissionServiceSpec.scala]: comment

File: app/models/ETag.scala (hunk 1 above)
Comment (vinnicombe) [reply app/models/ETag.scala]: https://github.com/hmrc/citizen-details#get-citizen-detailsninoetag

File: test/services/SubmissionServiceSpec.scala (hunk 3 above)
Comment (vinnicombe) [reply test/services/SubmissionServiceSpec.scala]: Uncommented

File: test/services/SubmissionServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+          result =>
+            result.length mustBe 2
+            result.contains(CurrentYearMinus1) mustBe true
+            result.contains(CurrentYearMinus2) mustBe true
+        }
+      }
+
+    }
+
+    "submitFRE" must {
```
Comment (MatthewAlexanderBeck) [inline test/services/SubmissionServiceSpec.scala]: submitFRE

File: test/services/TaiServiceSpec.scala [hunk 5]
```diff
@@ -90,6 +93,47 @@ class TaiServiceSpec extends SpecBase with MockitoSugar with ScalaFutures with I
         }
       }
     }
+
+    "updateFRE" when {
```
Comment (MatthewAlexanderBeck) [inline test/services/TaiServiceSpec.scala]: updateFRE

File: test/services/TaiServiceSpec.scala [hunk 6]
```diff
@@ -90,6 +93,47 @@ class TaiServiceSpec extends SpecBase with MockitoSugar with ScalaFutures with I
...
+          .thenReturn(Future.successful(HttpResponse(NO_CONTENT)))
+
+        val result = taiService.updatePsubAmount(fakeNino, taxYearInt, 100)
+
+        whenReady(result) {
+          _.status mustBe NO_CONTENT
+        }
+      }
+
+      "must exception on failed tai FRE update" in {
```
Comment (MatthewAlexanderBeck) [inline test/services/TaiServiceSpec.scala]: FRE update

File: app/services/TaiService.scala [hunk 7]
```diff
@@ -47,4 +51,24 @@ class TaiService @Inject()(taiConnector: TaiConnector) {
...
+
+    citizenDetailsConnector.getEtag(nino).flatMap {
+      response =>
+        response.status match {
+          case OK =>
+            Json.parse(response.body).validate[ETag] match {
+              case JsSuccess(body, _) =>
+                taiConnector.updateProfessionalSubscriptionAmount(nino, year, body.etag, grossAmount)
+              case JsError(e) =>
+                Logger.error(s"[TaiService.updateFRE][CitizenDetailsConnector.getEtag][Json.parse] failed $e")
```
[the remark this answers is not in the document]
Comment (vinnicombe) [reply app/services/TaiService.scala]: Ta!

File: test/services/SubmissionServiceSpec.scala (hunk 4 above)
Comment (vinnicombe) [reply test/services/SubmissionServiceSpec.scala]: Ta!

File: test/services/TaiServiceSpec.scala (hunk 5 above)
Comment (vinnicombe) [reply test/services/TaiServiceSpec.scala]: Ta!

File: test/services/TaiServiceSpec.scala (hunk 6 above)
Comment (vinnicombe) [reply test/services/TaiServiceSpec.scala]: Ta!

**Pull request merged**